File: android-app/app/src/main/java/com/example/androidapp/data/model/Movie.java
package com.example.androidapp.data.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.TypeConverters;

import java.util.List;

@Entity(tableName = "movies")
public class Movie {
    @PrimaryKey
    public int id;

    public String name;

    @TypeConverters(ListConverter.class)
    public List<Integer> categories;

    public int duration = 120;

    public String image = "uploads/movies/default-picture.jpg";

    public String video = "uploads/movies/default-video.mp4";

    public int ageLimit = 13;

    public String description = "No description available for this movie.";

    public Movie() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Integer> getCategories() {
        return categories;
    }

    public void setCategories(List<Integer> categories) {
        this.categories = categories;
    }

    public int getDuration() {
        return duration;
    }

    public void setDuration(int duration) {
        this.duration = duration;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getVideo() {
        return video;
    }

    public void setVideo(String video) {
        this.video = video;
    }

    public int getAgeLimit() {
        return ageLimit;
    }

    public void setAgeLimit(int ageLimit) {
        this.ageLimit = ageLimit;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: web-server/nodejs-server/models/movie.js
const mongoose = require("mongoose");

const Schema = mongoose.Schema;

const Movie = new Schema(
  {
    _id: {
      type: Number,
      required: true,
    },
    name: {
      type: String,
      required: true,
    },
    categories: {
      type: [{ type: Number, ref: "Category" }],
      default: [],
    },
    duration: {
      type: Number,
      default: 120,
    },
    image: {
      type: String,
      default: 'uploads/movies/default-picture.jpg',
    },
    video: {
      type: String,
      default: 'uploads/movies/default-video.mp4',
    },
    age_limit: {
      type: Number,
      default: 13,
    },
    description: {
      type: String,
      default: "No description available for this movie.",
    },
  }, { 
    versionKey: false,
    toJSON: { virtuals: true, transform: (doc, ret) => { delete ret._id; } },
    toObject: { virtuals: true, transform: (doc, ret) => { delete ret._id; } }
});

// Create a virtual field 'id' that returns the value of '_id'
Movie.virtual('id').get(function() {
    return this._id;
});

module.exports = mongoose.model("Movie", Movie);
